app/src/main/java/com/hyorim/sopt_assigmnet_1/SignUpActivity.kt

```diff
@@ -47,6 +46,50 @@ class SignUpActivity : AppCompatActivity() {
         return !isNameNull && !isIdNull && !isPwNull
     }
 
+    private fun initNetwork() {
+        val name = binding.nameEditText.text.toString()
+        val id = binding.IDEditText.text.toString()
+        val pw = binding.PWEditText.text.toString()
+
+        val requestSignUpData = RequestSignUpData(
+            binding.IDEditText.text.toString(),
+            binding.nameEditText.text.toString(),
+            binding.PWEditText.text.toString()
+        )
+
+        val call: Call<ResponseSignUpData> =
+            ServiceCreator.sampleService.postSignUp(requestSignUpData)
+
+        call.enqueue(object : Callback<ResponseSignUpData> {
+            override fun onResponse(
+                call: Call<ResponseSignUpData>,
+                response: Response<ResponseSignUpData>,
+            ) {
+                if (response.isSuccessful) {
+                    val data = response.body()?.data
+                    Toast.makeText(this@SignUpActivity,
+                        "${data?.name}님, 회원가입 성공",
+                        Toast.LENGTH_SHORT).show()
+
+                    // SignInActivity 로 이동
+                    var intent = Intent(this@SignUpActivity, SignInActivity::class.java)
+                    intent.putExtra("id", id)
+                        .putExtra("pw", pw)
```

Comment:
이 부분의 경우 scope함수인 apply를 활용해서

```suggestion
                    var intent = Intent(this@SignUpActivity, SignInActivity::class.java).apply {
                    putExtra("id", id)
                    putExtra("pw", pw)
                        }
```

이렇게 써도 된다잉~!! 참고로 apply의 경우 receiver로 받은 오브젝트를 리턴하는 특징을 갖고있어서 apply를 쓰는 거니께 참고하길 바란다잉~~!! 아래 scope함수의 특징 구별법 사진은 참고용으로 두고간다잉~~!!

<img src="https://t1.daumcdn.net/cfile/tistory/9925163E5DEBE18E01">

Reply:
아하~~ 구글링 해보면 여기저기서 apply 쓸때도 있고 intent.~~ 할때도 있어서 뭐가 다른가 궁금했는데!!
apply는 object return!! 하나 또 알아갑니다ㅎㅎㅎ